qa/workunits/rbd/rbd-nbd.sh

```diff
@@ -143,18 +149,37 @@ test ${blocks2} -eq ${blocks}
 # read-only option test
 _sudo rbd-nbd unmap ${DEV}
 DEV=`_sudo rbd-nbd map --read-only ${POOL}/${IMAGE}`
-_sudo rbd-nbd list-mapped | grep "^${DEV}$"
+PID=$(rbd-nbd list-mapped | awk -v pool=${POOL} -v img=${IMAGE} -v dev=${DEV} \
+    '$2 == pool && $3 == img && $5 == dev {print $1}')
+test -n "${PID}"
+ps -p ${PID} -o cmd | grep rbd-nbd
+
 _sudo dd if=${DEV} of=/dev/null bs=1M
 expect_false _sudo dd if=${DATA} of=${DEV} bs=1M oflag=direct
 _sudo rbd-nbd unmap ${DEV}
 
 # exclusive option test
 DEV=`_sudo rbd-nbd map --exclusive ${POOL}/${IMAGE}`
-_sudo rbd-nbd list-mapped | grep "^${DEV}$"
+PID=$(rbd-nbd list-mapped | awk -v pool=${POOL} -v img=${IMAGE} -v dev=${DEV} \
+    '$2 == pool && $3 == img && $5 == dev {print $1}')
+test -n "${PID}"
+ps -p ${PID} -o cmd | grep rbd-nbd
+
 _sudo dd if=${DATA} of=${DEV} bs=1M oflag=direct
 expect_false timeout 10 \
 	rbd bench ${IMAGE} --io-type write --io-size=1024 --io-total=1024
 _sudo rbd-nbd unmap ${DEV}
+
+# auto unmap test
+DEV=`_sudo rbd-nbd map ${POOL}/${IMAGE}`
+PID=$(rbd-nbd list-mapped | awk -v pool=${POOL} -v img=${IMAGE} -v dev=${DEV} \
+    '$2 == pool && $3 == img && $5 == dev {print $1}')
+test -n "${PID}"
+ps -p ${PID} -o cmd | grep rbd-nbd
+_sudo kill ${PID}
+sleep 1
+rbd-nbd list-mapped | expect_false grep "${DEV}"
```

Comment:
In the beginning of rbd-nbd.sh, there is:
`!/bin/bash -ex`
That means if there is any error happens during the shell excution, it will quit. "cleanup" is a function name defined in rbd-nbd.sh, in order to unmap nbd and rm image before quit.

So I am afraid:
`rbd-nbd list-mapped | expect_false grep ...`
This will cause the shell quit in first iteration. What is your opinion?